python/tmc5130.py

```diff
@@ -0,0 +1,1427 @@
+
+        # Chopper Config
+        chop_config.toff = 0x03  # 0010
+        chop_config.hstrt = 0x04
+        chop_config.hend = 0x1
+        chop_config.tbl = 0x2  # Blank time
+        chop_config.chm = 0x01  # chopper mode
+        chop_config.vhighfs = 0x01  # Enable Fullstep at high velocity
+        chop_config.vhighchm = 0x01  # High velocity chopper mode
+
+        # Cool Step Config - Stallguard
+        cool_conf.sgt = 0x00
+
+        # PWM Conf
+        pwm_mode_chop.pwm_autoscale = 1
+        pwm_mode_chop.freewheel = 0x1  # 0 default, 1 free wheeling
+        pwm_mode_chop.pwm_ampl = 0x255
+        pwm_mode_chop.pwm_grad = 0x4
+
+        tpwm_threshold.tpwmthrs = 500
+
+        # Move type : ramp mode
+        ramp_mode.ramp_mode = self.config.get("ramp_mode", 0)
+
+        # Write to TMC Driver
+        self._write(GeneralConfig.address, general_config.get_register())
+        self._write(ChopConf.address, chop_config.get_register())
+        self._write(CoolConf.address, cool_conf.get_register())
+        self._write(PWMConf.address, pwm_mode_chop.get_register())
+        self._write(TPWMTHRS.address, tpwm_threshold.get_register())
+        self._write(TCOOLTHRS.address, cool_threshold.get_register())
+        self._write(TPowerDown.address, power_down.get_register())
+        self._write(RampMode.address, ramp_mode.get_register())
+        self.set_current(**self.config.get("current_mA", {}))
+        self.set_target_position(0)
+        self.get_current_position(0)
+        self.set_motion_rpm(velocity_rpm=0, acceleration_rpm=0, deceleration_rpm=0)
+
+    def set_current(self, run_mA: int, hold_mA: int):
+        """
+        Set the current for the TMC Driver
+        """
+        Rsense = self.config.rsense
+        # Convert from mA to ihold and irun
+        irun = 32.0 * 1.41421 * run_mA / 1000.0 * (Rsense + 0.02) / 0.325 - 1
+        # Limit ihold to 31 and 0
+        irun = min(31, max(0, irun))
+        if hold_mA != 0:
+            ihold = 32.0 * 1.41421 * hold_mA / 1000.0 * (Rsense + 0.02) / 0.325 - 1
+        else:
+            ihold = 0
+        ihold = min(31, max(0, ihold))
+        idelay = 3
+
+        # Set the current
+        _current = IHoldIRun()
+        _current.ihold = int(ihold)
+        _current.irun = int(irun)
+        _current.idelay = int(idelay)
+
+        self._write(IHoldIRun.address, _current.get_register())
+
+    def set_motion_rpm(self, velocity_rpm: float, acceleration_rpm: float, deceleration_rpm: float):
+        """
+        Set the motion for the TMC Driver
+        """
+        velocity = self._from_rps(velocity_rpm / self.RPS_TO_RPM)
+        acceleration = self._from_rps2(acceleration_rpm / self.RPS_TO_RPM)
+        deceleration = self._from_rps2(deceleration_rpm / self.RPS_TO_RPM)
+        print(f"Velocity: {velocity} Acceleration: {acceleration} Deceleration: {deceleration}")
+        self.set_motion(velocity, acceleration, deceleration)
+
+    def set_motion_mms_degs(
```

Comment:
This is a confusing method. I sort of get what you're trying to do, but it needs to get a better name. 
```suggestion
    def set_motion_units(
```

Reply:
Done